src/main/java/com/znsio/teswiz/steps/Hooks.java

```diff
@@ -24,7 +24,8 @@ public void beforeScenario(Scenario scenario) {
                                   scenario.getName()));
         LOGGER.info(String.format("Hooks: Running test %s on %s", testExecutionContext.getTestName(),
                                   Runner.getPlatform().name()));
-        testExecutionContext.addTestState(TEST_CONTEXT.SCREENSHOT_MANAGER, new ScreenShotManager());
+        if(!Runner.getPlatform().name().equalsIgnoreCase("api"))
+            testExecutionContext.addTestState(TEST_CONTEXT.SCREENSHOT_MANAGER, new ScreenShotManager());
```

Comment:
This may lead to problems further on - when we have multi-user scenarios. Is there any particular reason why you have this conditional check?

Reply:
This was done with the aim to skip the step
**testExecutionContext.addTestState(TEST_CONTEXT.SCREENSHOT_MANAGER, new ScreenShotManager());**
as screenshot is not required in api test and thus screenshotDirectory was not initialised in **CucumberApiScanrioListener.java.**
hence this was giving exception.